iceoryx_hoofs/cmake/iceoryxversions.cmake

```diff
@@ -30,8 +30,8 @@ if(DEFINED ${PROJECT_VERSION_TWEAK})
   set(IOX_VERSION_TWEAK ${PROJECT_VERSION_TWEAK})
 endif()
 
-configure_file("${CMAKE_CURRENT_SOURCE_DIR}/cmake/iceoryx_versions.hpp.in"
-  "${CMAKE_BINARY_DIR}/generated/iceoryx/include/iceoryx_versions.hpp" @ONLY)
-install(FILES ${CMAKE_BINARY_DIR}/generated/iceoryx/include/iceoryx_versions.hpp
+configure_file("${CMAKE_CURRENT_SOURCE_DIR}/cmake/iceoryx_versions.h.in"
+  "${CMAKE_BINARY_DIR}/generated/iceoryx_hoofs/include/iceoryx_versions.h" @ONLY)
```

Comment:
@elBoberido It's a bit odd to me that the version header now lives in `iceoryx_hoofs`. Wouldn't this fit better into `iceoryx_platform`?

Reply:
Best would be below platform since the platform can be changed by the user. `hoofs` is currently the next best thing.

Reply:
There is a `IOX_PLATFORM_PATH` cmake argument for custom platforms. If the version header would be in the platform it would be overwritten with the one from the custom platform.

I once had an idea to simplify the platform with C++17. Once that is implemented it would be possible